### PR TITLE
Update minigraph.py to parse kubernetes config from minigraph.xml

### PR DESCRIPTION
**Pull request by renukamanavalan**
**- Why I did it**
To make  the kubernetes server IP & enabled/disabled status configurable through minigraph.
Saved cluster name into "DEVICE_METADATA|localhost" , so this can be available for kubernetes use.

**- How I did it**
Updated the parser (minigraph.py) to get it from MetadataDeclaration/Devices/DeviceMetadata section for this device.
Add to "DEVICE_METADATA|localhost" , if ClusterName is available in PngDec/Devices/Device for localhost

**- How to verify it**
Unit test is added to parse & test out.
For manual test add the following to MetadataDeclaration/Devices/DeviceMetadata section. call "config load_minigraph" and verify through "show kube server config"

Look for "DEVICE_METADATA": { "localhost": { ... "cluster": "<this device cluster name", ...}}, if minigraph.xml does carry it in JSON dump or /etc/sonic/config_db.json

```
<a:DeviceProperty>
<a:Name>KubernetesEnabled</a:Name>
<a:Reference i:nil="true"/>
<a:Value>0</a:Value>
</a:DeviceProperty>
<a:DeviceProperty>
<a:Name>KubernetesServerIp</a:Name>
<a:Reference i:nil="true"/>
<a:Value>10.10.10.10</a:Value>
</a:DeviceProperty>
```
**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [X] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by kcudnik — DISMISSED** [review]

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -808,7 +809,11 @@ def parse_meta(meta, hname):
                     cloudtype = value
                 elif name == "ResourceType":
                     resource_type = value
-    return syslog_servers, dhcp_servers, ntp_servers, tacacs_servers, mgmt_routes, erspan_dst, deployment_id, region, cloudtype, resource_type
+                elif name == "KubernetesEnabled":
+                    kube_data["disable"] = value
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:813]: >disable [](start = 31, length = 7)

I am confused, do you mean "enabled" ?

**Comment by qiluo-msft** [comment]
> 

Remove extra blank lines

---
Refers to: src/sonic-config-engine/tests/test_minigraph_case.py:300 in bab6ca0. [](commit_id = bab6ca08c5a1e455715843598d0fbb87e0c61dea, deletion_comment = False)

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -1174,6 +1180,13 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         }
     }
 
+    if kube_data:
+        results['KUBERNETES_MASTER'] = { 'SERVER': {
+            'disable': 'True' if kube_data.get('disable', '0') == '0' else 'False',
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:1185]: >'True' [](start = 23, length = 6)

If you need True/False string, you can use str(bool)

Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:1185]: >'di [](start = 12, length = 3)

Indent inner block one more level?

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (renukamanavalan) [reply src/sonic-config-engine/minigraph.py]: Name changed to be more appropriate. Good catch. No logical code change still.

File: src/sonic-config-engine/minigraph.py (hunk 2 above)
Comment (renukamanavalan) [reply src/sonic-config-engine/minigraph.py]: Yep, a better one. Taken.

Comment (renukamanavalan) [reply src/sonic-config-engine/minigraph.py]: This is the default indent provided by vi editor for .py script for a dict object.

**Comment by renukamanavalan** [comment]
> > 
> 
> Remove extra blank lines
> 
> Refers to: src/sonic-config-engine/tests/test_minigraph_case.py:300 in [bab6ca0](https://github.com/Azure/sonic-buildimage/commit/bab6ca08c5a1e455715843598d0fbb87e0c61dea). [](commit_id = [bab6ca0](https://github.com/Azure/sonic-buildimage/commit/bab6ca08c5a1e455715843598d0fbb87e0c61dea), deletion_comment = False)

Can you please tell mem which file & line number ?
If it is last line in tests/test_minigraph_case.py, I have removed it.

File: src/sonic-config-engine/minigraph.py (hunk 2 above)
Comment (qiluo-msft) [reply src/sonic-config-engine/minigraph.py]: I checked with a python formatter.
```python
    if kube_data:
        results["KUBERNETES_MASTER"] = {
            "SERVER": {
                "disable": "True" if kube_data.get("disable", "0") == "0" else "False",
                "disable": str(kube_data.get("enable", "0") == "0"),
                "ip": kube_data.get("ip", ""),
            }
        }
```

---
In reply to: [568229445](https://github.com/Azure/sonic-buildimage/pull/6633#discussion_r568229445) [](ancestors = 568229445)

**Review by qiluo-msft — COMMENTED** [review]
Minor format issue. Otherwise lgtm.

File: src/sonic-config-engine/minigraph.py (hunk 2 above)
Comment (renukamanavalan) [reply src/sonic-config-engine/minigraph.py]: Got it now. Done

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**